### PR TITLE
Fix some bugs in the inflight-message code

### PR DESCRIPTION
**Pull request by osheroff**
Was trying some of this code out in a real way, trying to trace down some out-of-memory errors.  Found a couple of bugs that need fixing.

@zendesk/rules

### CODE REVIEW

**Comment by osheroff** [comment]
cc @dasch

File: src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaProducer.java [hunk 1]
```diff
@@ -45,9 +45,11 @@ public void onCompletion(RecordMetadata md, Exception e) {
 		if ( e != null ) {
 			if ( e instanceof RecordTooLargeException ) {
 				LOGGER.error("RecordTooLargeException @ " + position + " -- " + key);
-				LOGGER.error("Maxwell dropped a row because it was too large (" + e.getLocalizedMessage() + ")");
+				LOGGER.error(e.getLocalizedMessage());
 				LOGGER.error("Considering raising max.request.size broker-side.");
-				inflightMessages.completeMessage(position);
+
+				if ( isTXCommit )
+					inflightMessages.completeMessage(position);
```
Comment (dadah89) [inline src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaProducer.java]: don't you need to set the new position in `context`? (similar to what is done below)

**Comment by dasch** [comment]
Are multiple threads writing to the inflight message tracker concurrently? It would be interesting to see if you get congestion.

:+1:

File: src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaProducer.java (hunk 1 above)
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaProducer.java]: i do.  will fix.

**Comment by osheroff** [comment]
@dasch the callback is executed on the kafka producer thread, so that can write concurrently with the maxwell producer

**Comment by osheroff** [comment]
@dadah89 fixed comments

**Pull request merged**